tests/integration/ros3/test_ros3.py

```diff
@@ -0,0 +1,52 @@
+from pynwb import NWBHDF5IO
+from pynwb.testing import TestCase
+import requests
+
+
+def _search_dandi_assets(url, filepath):
+    response = requests.request("GET", url, headers={"Accept": "application/json"}).json()
+
+    for asset in response["results"]:
+        if filepath == asset["path"]:
+            return asset["asset_id"]
+
+    if response.get("next", None):
+        return _search_dandi_assets(response["next"], filepath)
+
+    return None
+
+
+def get_dandi_asset_id(dandiset_id, filepath):
+    url = f"https://api.dandiarchive.org/api/dandisets/{dandiset_id}/versions/draft/assets/"
+    asset_id = _search_dandi_assets(url, filepath)
+    if asset_id is None:
+        raise ValueError(f'path {filepath} not found in dandiset {dandiset_id}.')
+    return asset_id
+
+
+def get_dandi_s3_url(dandiset_id, filepath):
+    """Get the s3 location for any NWB file on DANDI"""
+
+    asset_id = get_dandi_asset_id(dandiset_id, filepath)
+    url = f"https://api.dandiarchive.org/api/dandisets/{dandiset_id}/versions/draft/assets/{asset_id}/download/"
+
+    s3_url = requests.request(url=url, method='head').url
+    if '?' in s3_url:
+        return s3_url[:s3_url.index('?')]
+    return s3_url
```

Comment:
I think those functions may make sense to put into ``pynwb.testing`` instead

Reply:
Yeah, I agree. Might be useful for others when writing tests for extensions and such.

Reply:
I have now done that